pkg/api/registry/registry.proto

```diff
@@ -22,61 +22,41 @@ message Destination {
     uint32 weight = 2;
 }
 
-message NetworkServiceManager {
-    string name = 1;
-    string url = 2;
-    google.protobuf.Timestamp expiration_time = 3;
-    string state = 4;
-}
-
 message NetworkServiceEndpoint {
     string name = 1;
     string payload = 2;
     string network_service_name = 3;
-    string network_service_manager_name = 4;
-    map<string, string> labels = 5;
-    string state = 6;
-
+    map<string, string> labels = 4;
+    string state = 5;
     // Endpoint callback address, could be any of unix/tcp/callback schema ones.
-    string url = 7;
-}
-
-message FindNetworkServiceRequest {
-    string network_service_name = 1;
-}
-
-message FindNetworkServiceResponse {
-    string payload = 1;
-    NetworkService network_service = 2;
-    map<string, NetworkServiceManager> network_service_managers = 3;
-    repeated NetworkServiceEndpoint network_service_endpoints = 4;
+    string url = 6;
+    // Network service manager URL for the endpoint.
+    string network_service_manger_url = 7;
+    google.protobuf.Timestamp expiration_time = 8;
 }
 
-message NSERegistration {
-    NetworkService network_service = 1;
-    NetworkServiceManager network_service_manager = 2;
-    NetworkServiceEndpoint network_service_endpoint = 3;
-}
-
-message RemoveNSERequest {
-    string network_service_endpoint_name = 1;
+service NetworkServiceEndpointRegistry {
+    rpc RegisterOrUpdate (NetworkServiceEndpoint) returns (NetworkServiceEndpoint);
+    rpc Monitor (NetworkServiceEndpointSelector) returns (stream NetworkServiceEndpoint);
+    rpc Find (NetworkServiceEndpointSelector) returns (stream NetworkServiceEndpoint);
```

Comment:
What is the difference between Monitor and Find here? :)

Reply:
In my mind, `Monitor` will be implemented as a `while(true)` goroutine that will notify about changes. 
`Find` should just send all known entries and close the stream.